### PR TITLE
Add luacheck

### PR DESCRIPTION
**Pull request by h-michael**
- [x] define .luacheckrc
- [x] set Github Actions
- [x] fix luacheck error

### CODE REVIEW

File: lua/nvim_lsp/pyls.lua [hunk 1]
```diff
@@ -1,5 +1,4 @@
 local skeleton = require 'nvim_lsp/skeleton'
-local util = require 'nvim_lsp/util'
```
Comment (h-michael) [inline lua/nvim_lsp/pyls.lua]: unused

File: lua/nvim_lsp/texlab.lua [hunk 2]
```diff
@@ -22,14 +22,14 @@ end
...
-  end)
-end
+-- local function build_cancel_all(bufnr)
+--   bufnr = util.validate_bufnr(bufnr)
+--   local params = { token = "texlab-build-*" }
+--   lsp.buf_request(bufnr, 'window/progress/cancel', params, function(err, method, result, client_id)
+--     if err then error(tostring(err)) end
+--     print("Cancel result", vim.inspect(result))
+--   end)
+-- end
```
Comment (h-michael) [inline lua/nvim_lsp/texlab.lua]: I don't know whether we can remove this unused function or not.

File: lua/nvim_lsp/util.lua [hunk 3]
```diff
@@ -14,19 +14,19 @@ function M.validate_bufnr(bufnr)
   return bufnr == 0 and api.nvim_get_current_buf() or bufnr
 end
 
-function M.add_hook_before(fn, new_fn)
-  if fn then
+function M.add_hook_before(func, new_fn)
```
Comment (h-michael) [inline lua/nvim_lsp/util.lua]: shadowing: https://github.com/neovim/nvim-lsp/pull/86/files#diff-e23e94d10ef42d80a01f1577ed178ce9R6

File: lua/nvim_lsp/vimls.lua [hunk 4]
```diff
@@ -1,7 +1,6 @@
 local skeleton = require "nvim_lsp/skeleton"
 local util = require "nvim_lsp/util"
 local lsp = vim.lsp
-local api = vim.api
```
Comment (h-michael) [inline lua/nvim_lsp/vimls.lua]: unused

File: scripts/docgen.lua [hunk 5]
```diff
@@ -178,9 +178,9 @@ local function make_lsp_sections()
                   -- The list of properties.
                   make_section(0, '\n\n', sorted_map_table(default_settings.properties, function(k, v)
                     local function tick(s) return string.format("`%s`", s) end
-                    local function pre(s) return string.format("<pre>%s</pre>", s) end
-                    local function code(s) return string.format("<code>%s</code>", s) end
                     local function bold(s) return string.format("**%s**", s) end
+                    -- local function pre(s) return string.format("<pre>%s</pre>", s) end
+                    -- local function code(s) return string.format("<code>%s</code>", s) end
```
Comment (h-michael) [inline scripts/docgen.lua]: unused, But do we use these functions in the future?

File: scripts/docgen.lua [hunk 6]
```diff
@@ -10,10 +10,10 @@ local function template(s, params)
   return (s:gsub("{{([^{}]+)}}", params))
 end
 
-local function map_list(t, fn)
+local function map_list(t, func)
```
Comment (h-michael) [inline scripts/docgen.lua]: shadowing: https://github.com/neovim/nvim-lsp/pull/86/files#diff-173291637905cca10e5a7751c4f4f4ccR6

File: .github/workflows/lint.yml [hunk 7]
```diff
@@ -0,0 +1,22 @@
+name: lint
+
+on:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  lint:
```
Comment (justinmk) [inline .github/workflows/lint.yml]: why not add this as a job in the `jobs` list of `main.yml`?

**Comment by justinmk** [comment]
LGTM after comments

File: .github/workflows/lint.yml (hunk 7 above)
Comment (h-michael) [reply .github/workflows/lint.yml]: This job needs to run on pull request. But existing job needs to run on master push.
So we need separate these jobs into other workflows.

**Pull request merged**